src/client/dfs/duns.c

```diff
@@ -103,10 +345,109 @@ duns_resolve_path(const char *path, struct duns_attr_t *attr)
 	return 0;
 }
 
-/*
- * TODO: should detect backend file system and if Lustre, use special LOV/LVM EA
- * instead of regular extended attribute.
- */
+static int
+duns_create_lustre_path(daos_handle_t poh, const char *path,
+			struct duns_attr_t *attrp)
+{
+	char			pool[37], cont[37];
+	char			oclass[10], type[10];
+	char			str[DUNS_MAX_XATTR_LEN + 1];
+	int			len;
+	int			try_multiple = 1;		/* boolean */
+	daos_handle_t		coh;
+	daos_cont_info_t	co_info;
+	dfs_t			*dfs;
+	int			rc, rc2;
+
+	/* XXX pool must already be created, and associated DFuse-mount
+	 * should already be mounted
+	 */
+
+	/* XXX if liblustreapi is not binded, do it now ! */
+	if (liblustre_binded == false && liblustre_notfound == false) {
+		rc = bind_liblustre();
+		if (rc)
+			return -DER_INVAL;
+	}
+
+	uuid_unparse(attrp->da_puuid, pool);
+	daos_oclass_id2name(attrp->da_oclass_id, oclass);
+	daos_unparse_ctype(attrp->da_type, type);
+
+	/* create container with specified container uuid (try_multiple=0)
+	 * or a generated random container uuid (try_multiple!=0).
+	 */
+	if (!uuid_is_null(attrp->da_cuuid)) {
+		try_multiple = 0;
+		uuid_unparse(attrp->da_cuuid, cont);
+		D_INFO("try create once with provided container UUID: %36s\n",
+			cont);
+	}
+	/* create container */
+	do {
+		if (try_multiple) {
+			uuid_generate(attrp->da_cuuid);
+			uuid_unparse(attrp->da_cuuid, cont);
+		}
+
+		if (attrp->da_type == DAOS_PROP_CO_LAYOUT_POSIX) {
+			dfs_attr_t	dfs_attr;
+
+ 			/** TODO: set Lustre FID here. */
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs